backend/src/lib/report-generator.js

```diff
@@ -147,6 +190,121 @@ expressions.filters.changeID = function (input, prefix) {
     return input.replace("IDX-", prefix);
 }
 
+// Default value: returns input if it is truthy, otherwise its parameter.
+// Example producing a comma-separated list of affected systems, falling-back on the whole audit scope: {affected | lines | d: (scope | select: 'name') | join: ', '}
+expressions.filters.d = function(input, s) {
+    return (input && input != "undefined") ? input : s;
+}
+
+// Display "From ... to ..." dates nicely, removing redundant information when the start and end date occur during the same month or year: {date_start | fromTo: date_end:'fr' | capfirst}
+// To internationalize or customize the resulting string, associate the desired output to the strings "from {0} to {1}" and "on {0}" in your Pwndoc translate file.
+expressions.filters.fromTo = function(start, end, locale) {
```

Comment:
`convertDateLocale` and `fromTo` really have two different use.

Maybe the documentation of the `fromTo` filter was not very clear so I updated it in [74c6407](https://github.com/pwndoc/pwndoc/pull/281/commits/74c6407a1938ceb9809df6ca1e48ec49daf57ed5).

The goal is to have the date format being handled semi-automatically to easily get more natural start and end dates in the reports :

```
- Same day        -> On 28/11/2022
- Same month      -> From 28 to 30/11/2022
- Same year       -> From 28/11 to 02/12/2022
- Different years -> From 28/11/2022 to 06/01/2023
```

While I suppose it should be possible to reimplement this purely within the template by relying on multiple calls to `convertDateLocale`, date comparisons and a potential additional style option, this would require something like a whole page of unreadable code embedded in the template each time a start and end date should be displayed. This sound to me as clearly ineffective and hard to main, and I also suppose that displaying start and end is part of most if not all reports so such a mechanism might be useful.

That's why I supposed that such filter would be useful to the community, but if it's not feel free to delete it and I will keep mine as part of my custom filters.